server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -852,19 +859,31 @@ private static IndexNotFoundException indexNotFoundException(String expression)
         }
 
         private boolean isEmptyOrTrivialWildcard(List<String> expressions) {
-            return expressions.isEmpty() || (expressions.size() == 1 && (MetaData.ALL.equals(expressions.get(0)) ||
-                Regex.isMatchAllPattern(expressions.get(0))));
+            if (expressions.isEmpty()) {
+                return true;
+            } else if (expressions.size() == 1) {
+                final String expression = expressions.get(0);
+                return MetaData.ALL.equals(expression) || Regex.isMatchAllPattern(expression);
+            } else {
+                return false;
+            }
```

Comment:
While this change isn't related to this PR directly, it's a great readability improvement - I thought there was a behavior difference between the new and old versions until I looked very closely at the parens in the old version.